### ISSUE TITLE
Diagram was not generated for postgres views

### ISSUE THREAD

**Issue by ChubbyCub** [schemaspy/schemaspy#1207]
I am running schemaspy with the followings:
1. schemaspy-6.2.3.jar
2. postgresql-42.2.22.jar
3. dot - graphviz version 8.0.5 (20230430.1635)
4. macOS 13.3.1 (a) (22E772610a)
5. openjdk version "1.8.0_272"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_272-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.272-b10, mixed mode)

My schemaspy config file is like below:
schemaspy.t=pgsql
schemaspy.dp=/Users/username/Development/postgresql-42.2.22.jar
schemaspy.host=localhost
schemaspy.port=5432
schemaspy.db=[dbname]
schemaspy.u=[dbuser]
schemaspy.p=[dbpassword]
schemaspy.o=/Users/username/Development/spy
schemaspy.s=[schemaname]
viewTypes=VIEW,MATERIALIZED VIEW

After running this command `java -jar schemaspy-6.2.3.jar -configFile schemaspy.properties`, schemaspy was running. But the run kept spitting out errors like this:

`ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: in label of node labels
ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: in label of node debtor
ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.compact.dot -orelationships.real.compact.png -Tcmapx: in label of node client
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: in label of node labels
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: in label of node debtor
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo relationships.real.large.dot -orelationships.real.large.png -Tcmapx: in label of node client`

When I open up the `index.html` file, I did not see any diagrams for the views. It only includes the relationships between tables. When inspecting the `views` tab, it was able to detect potentially referenced views/tables.

If anyone ran into the same issue, please let me know.

**Comment by npetzall** [comment]
Hi, do you have any other errors in the output. The included ones are tricky for us to handle, since we get them from standard error stream. But in this case they are only warnings.

If you are running against postgres 11 or newer you should be getting an error in the beginning which can be avoided by using `pgsql11` instead of `pgsql` as value for `schemaspy.t`

But since you get potential references, you should get definition and there should be a diagram in that specific view page.

**Comment by ChubbyCub** [comment]
`INFO  - Writing/diagramming details
ERROR - dot -Tpng:cairo client_debtor.1degree.dot -oclient_debtor.1degree.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo client_debtor.1degree.dot -oclient_debtor.1degree.png -Tcmapx: in label of node client
ERROR - dot -Tpng:cairo client_debtor.1degree.dot -oclient_debtor.1degree.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo client_debtor.1degree.dot -oclient_debtor.1degree.png -Tcmapx: in label of node debtor
ERROR - dot -Tpng:cairo client_debtor.2degrees.dot -oclient_debtor.2degrees.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo client_debtor.2degrees.dot -oclient_debtor.2degrees.png -Tcmapx: in label of node client
ERROR - dot -Tpng:cairo client_debtor.2degrees.dot -oclient_debtor.2degrees.png -Tcmapx: Warning: cell size too small for content
ERROR - dot -Tpng:cairo client_debtor.2degrees.dot -oclient_debtor.2degrees.png -Tcmapx: in label of node debtor
..ERROR - dot -Tpng:cairo client_contract.1degree.dot -oclient_contract.1degree.png -Tcmapx: Warning: cell size too small for content`

Here's an example of other similar errors being logged. 

Yes - I got the definition, and there's a diagram of the view, but there's no implied relationships (like the underlying views or tables that were fed into the view).

**Comment by npetzall** [comment]
That's the same type of errors which are warnings.

This might be an enhancement request or something that we can improve. The implied functionality works just by comparing column name and type. Since we don't do any real view definition parsing, I believe the potential references is basically just checking if any table/view name exist in view definition.

So a view that renames a column from the source wouldn't be considered implied. Names don't match. But if view definition where processed a bit more like handling the rename which often is an `as`, implied would work. Since we have potential references we have a short list of possible implied tables to match against.

Is this possibly the case does the view definition rename columns?

**Comment by ChubbyCub** [comment]
For example, I have this view definition:

`SELECT client.uuid,
    client.division_uuid,
    client.is_approved_and_configured,
    client.is_manually_created,
    current_client_contract.credit_limit,
    current_client_contract.product_type,
    balance_view.ar_balance,
    balance_view.outstanding_balance
   FROM (((capfin.client
     LEFT JOIN ( SELECT DISTINCT ON (current_client_contract_view.client_uuid) current_client_contract_view.uuid,
            current_client_contract_view.client_uuid
           FROM capfin.current_client_contract_view
          ORDER BY current_client_contract_view.client_uuid, current_client_contract_view.as_of_date DESC NULLS LAST, current_client_contract_view.effective_date DESC NULLS LAST) current_client_contract_uuid ON ((current_client_contract_uuid.client_uuid = client.uuid)))
     LEFT JOIN capfin.client_contracts_view current_client_contract ON ((current_client_contract_uuid.uuid = current_client_contract.uuid)))
     JOIN capfin.client_balance_summary_view balance_view ON ((client.uuid = balance_view.client_uuid)));`

The program correctly identified references to client table and current_client_contract_view. However, on the relationship tab where the relationship diagram is being displayed, I do not see this view there. Same with other views that the program identified. So I guess this is not an existing feature of the program then.

**Comment by npetzall** [comment]
I don't think so, but for this use-case as they are not renamed only prefixed, it might be simpler to handle.

I'll look into it, but I would say it's a missing feature.

**Comment by npetzall** [comment]
I wonder if a quick would be to have diagram more as lineage than actually connecting columns, from the visualisation that might be enough.

You know that the table is "included" in the view, exactly which column could be added later.
